quanda/utils/datasets/dataset_handlers.py

```diff
@@ -0,0 +1,302 @@
+
+        Parameters
+        ----------
+        batch : Dict[str, torch.Tensor]
+            The batch dictionary containing inputs and labels.
+        device : Union[str, torch.device]
+            The device to move the tensors to.
+
+        Returns
+        -------
+        Tuple[Dict[str, torch.Tensor], torch.Tensor]
+            The processed inputs dictionary and labels on the specified device.
+
+        """
+        inputs = {k: v.to(device) for k, v in batch.items() if k != "labels"}
+        labels = batch["labels"].to(device)
+        return inputs, labels
+
+    def get_model_inputs(
+        self, inputs: Dict[str, torch.Tensor]
+    ) -> Dict[str, torch.Tensor]:
+        """Extract model inputs from the processed batch inputs.
+
+        Parameters
+        ----------
+        inputs : Dict[str, torch.Tensor]
+            The processed batch inputs dictionary.
+
+        Returns
+        -------
+        Dict[str, torch.Tensor]
+            The inputs to be passed to the model.
+
+        """
+        allowed_keys = {"input_ids", "attention_mask", "token_type_ids"}
+        return {
+            key: value for key, value in inputs.items() if key in allowed_keys
+        }
+
+    def extract_predictions(self, outputs: Any) -> torch.Tensor:
+        """Extract predictions from model outputs.
+
+        Parameters
+        ----------
+        outputs : Any
+            The model outputs.
+
+        Returns
+        -------
+        torch.Tensor
+            The extracted predictions.
+
+        """
+        return outputs.logits.argmax(dim=-1)
+
+    def create_dataloader(
+        self, dataset: datasets.Dataset, batch_size: int, shuffle: bool = False
+    ) -> DataLoader:
+        """Create a DataLoader for the dataset.
+
+        Parameters
+        ----------
+        dataset : datasets.Dataset
+            The dataset to load.
+        batch_size : int
+            The batch size to use.
+        shuffle : bool, optional
+            Whether to shuffle the data, by default False.
+
+        Returns
+        -------
+        DataLoader
+            Configured DataLoader.
+
+        """
+        return DataLoader(
+            dataset,
+            batch_size=batch_size,
+            collate_fn=default_data_collator,
+            shuffle=shuffle,
+        )
+
+
+def get_dataset_handler(
+    dataset: Union[torch.utils.data.Dataset, datasets.Dataset],
+) -> DatasetHandler:
+    """Return the correct DatasetHandler for the given dataset.
+
+    Parameters
+    ----------
+    dataset : Union[torch.utils.data.Dataset, datasets.Dataset]
+        The dataset which is either a PyTorch Dataset or HuggingFace Dataset.
+
+    Returns
+    -------
+    DatasetHandler
+        A handler instance suited for the dataset.
+
+    Raises
+    ------
+    ValueError
+        If the dataset is not supported.
+
+    """
+    if isinstance(dataset, datasets.Dataset):
+        return HuggingFaceDatasetHandler()
```

Comment:
Here we could already check if a dataset item contains "labels" key, and throw an error if it doesn't (for some early prevention)